### PR TITLE
merge tasks from liron-ecs-integration to master

### PR DESCRIPTION
**Pull request by Lironrad**
Added support batch queue in ecs integration under aws elastigroup.
Refactor spot percentage risk from float to int.
Added block device mapping to ocean ecs.

### CODE REVIEW

File: service/ocean/providers/aws/launch_spec_ecs.go [hunk 1]
```diff
@@ -70,6 +71,39 @@ type ECSAutoScaleHeadroom struct {
 	nullFields      []string
 }
 
+type ECSBlockDeviceMapping struct {
+	DeviceName  *string `json:"deviceName,omitempty"`
+	NoDevice    *string `json:"noDevice,omitempty"`
+	VirtualName *string `json:"virtualName,omitempty"`
+	ECSEBS      *ECSEBS `json:"ebs,omitempty"`
```
Comment (liranp) [inline service/ocean/providers/aws/launch_spec_ecs.go:78]:
```suggestion
	EBS      *ECSEBS `json:"ebs,omitempty"`
```

File: service/ocean/providers/aws/launch_spec_ecs.go [hunk 2]
```diff
@@ -70,6 +71,39 @@ type ECSAutoScaleHeadroom struct {
...
+
+type ECSEBS struct {
+	DeleteOnTermination  *bool                 `json:"deleteOnTermination,omitempty"`
+	Encrypted            *bool                 `json:"encrypted,omitempty"`
+	KMSKeyID             *string               `json:"kmsKeyId,omitempty"`
+	SnapshotID           *string               `json:"snapshotId,omitempty"`
+	VolumeType           *string               `json:"volumeType,omitempty"`
+	IOPS                 *int                  `json:"iops,omitempty"`
+	VolumeSize           *int                  `json:"volumeSize,omitempty"`
+	ECSDynamicVolumeSize *ECSDynamicVolumeSize `json:"dynamicVolumeSize,omitempty"`
```
Comment (liranp) [inline service/ocean/providers/aws/launch_spec_ecs.go:92]:
```suggestion
	DynamicVolumeSize *ECSDynamicVolumeSize `json:"dynamicVolumeSize,omitempty"`
```

File: service/ocean/providers/aws/launch_spec_ecs.go [hunk 3]
```diff
@@ -420,3 +461,138 @@ func (o *ECSAutoScaleHeadroom) SetNumOfUnits(v *int) *ECSAutoScaleHeadroom {
...
+
+func (o *ECSBlockDeviceMapping) SetVirtualName(v *string) *ECSBlockDeviceMapping {
+	if o.VirtualName = v; o.VirtualName == nil {
+		o.nullFields = append(o.nullFields, "VirtualName")
+	}
+	return o
+}
+
+func (o *ECSBlockDeviceMapping) SetEBS(v *ECSEBS) *ECSBlockDeviceMapping {
+	if o.ECSEBS = v; o.ECSEBS == nil {
```
Comment (liranp) [inline service/ocean/providers/aws/launch_spec_ecs.go:495]:
```suggestion
	if o.EBS = v; o.EBS == nil {
```

File: service/ocean/providers/aws/launch_spec_ecs.go [hunk 4]
```diff
@@ -420,3 +461,138 @@ func (o *ECSAutoScaleHeadroom) SetNumOfUnits(v *int) *ECSAutoScaleHeadroom {
...
+
+func (o *ECSEBS) SetVolumeSize(v *int) *ECSEBS {
+	if o.VolumeSize = v; o.VolumeSize == nil {
+		o.nullFields = append(o.nullFields, "VolumeSize")
+	}
+	return o
+}
+
+func (o *ECSEBS) SetDynamicVolumeSize(v *ECSDynamicVolumeSize) *ECSEBS {
+	if o.ECSDynamicVolumeSize = v; o.ECSDynamicVolumeSize == nil {
```
Comment (liranp) [inline service/ocean/providers/aws/launch_spec_ecs.go:561]:
```suggestion
	if o.DynamicVolumeSize = v; o.DynamicVolumeSize == nil {
```

File: spotinst/version.go [hunk 5]
```diff
@@ -1,7 +1,7 @@
 package spotinst
 
 // SDKVersion is the current version of the SDK.
-const SDKVersion = "1.61.0"
+const SDKVersion = "1.62.0"
```
Comment (liranp) [inline spotinst/version.go:4]:
```suggestion
const SDKVersion = "1.61.0"
```

The version number is bumped for every new release as part of the release process.

File: service/elastigroup/providers/aws/aws.go [hunk 6]
```diff
@@ -478,7 +486,7 @@ type Predictive struct {
 }
 
 type Strategy struct {
-	Risk                     *float64         `json:"risk,omitempty"`
+	Risk                     *int             `json:"risk,omitempty"`
```
Comment (liranp) [inline service/elastigroup/providers/aws/aws.go:489]: This change breaks the backward compatibility of the SDK. Is there a legitimate reason for the change?

File: service/ocean/providers/aws/launch_spec_ecs.go [hunk 7]
```diff
@@ -70,6 +71,39 @@ type ECSAutoScaleHeadroom struct {
 	nullFields      []string
 }
 
+type ECSBlockDeviceMapping struct {
+	DeviceName  *string `json:"deviceName,omitempty"`
+	NoDevice    *string `json:"noDevice,omitempty"`
+	VirtualName *string `json:"virtualName,omitempty"`
+	EBS      *ECSEBS `json:"ebs,omitempty"`
```
Comment (liranp) [inline service/ocean/providers/aws/launch_spec_ecs.go:78]: Please re-format the code.

File: service/ocean/providers/aws/launch_spec_ecs.go [hunk 8]
```diff
@@ -70,6 +71,39 @@ type ECSAutoScaleHeadroom struct {
...
+
+type ECSEBS struct {
+	DeleteOnTermination  *bool                 `json:"deleteOnTermination,omitempty"`
+	Encrypted            *bool                 `json:"encrypted,omitempty"`
+	KMSKeyID             *string               `json:"kmsKeyId,omitempty"`
+	SnapshotID           *string               `json:"snapshotId,omitempty"`
+	VolumeType           *string               `json:"volumeType,omitempty"`
+	IOPS                 *int                  `json:"iops,omitempty"`
+	VolumeSize           *int                  `json:"volumeSize,omitempty"`
+	DynamicVolumeSize *ECSDynamicVolumeSize `json:"dynamicVolumeSize,omitempty"`
```
Comment (liranp) [inline service/ocean/providers/aws/launch_spec_ecs.go:92]: Same here.

File: service/elastigroup/providers/aws/aws.go (hunk 6 above)
Comment (omerspot) [reply service/elastigroup/providers/aws/aws.go:489]: @liranp 
The reason was type mismach with the optimizer that was reflected eventually to TF, it is possible to fix on provider side and cast before sending.
Would it be a more preferable solution to you?
@Lironrad

**Pull request closed without merging**